### PR TITLE
feat(storybook): use native storybook/angular executor

### PR DESCRIPTION
**Pull request by mandarini**
## Current Behavior
Right now, when we're invoking the Storybook target, the `@nrwl/storybook:storybook` runs, which is basically a wrapper around `buildDevStandalone` of Storybook. This may lead to inconsistencies between the native Storybook result and the Nx result. Plus, it brings the need for maintenance.

## Expected Behavior

If this works out, in the future, we're hoping to be using the native Storybook executors to serve and build Storybook, instead of the nx-wrapped ones, for *Angular* projects.

eg. 

```
    "storybook": {
      "executor": "@storybook/angular:start-storybook",
      ....
```

#### Transform your existing Angular project `storybook` and `build-storybook` targets

Use the generator `@nrwl/storybook:change-storybook-targets` to change your existing Angular project Storybook targets to use the native executor.

Then run the storybook executor as you would normally (`nx storybook <PROJECT-NAME>`)

#### Generate Storybook Configuration for a new Angular project

The new configuration will have the new targets

## Sample Repo

Check out [this repository](https://github.com/mandarini/using-native-executor) - it shows all use cases.

### CODE REVIEW

File: packages/devkit/src/executors/parse-target-string.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ import type { Target } from 'nx/src/commands/run';
  */
 export function parseTargetString(targetString: string): Target {
   const [project, target, configuration] = targetString.split(':');
+  console.log('KATERINA', project, target, configuration);
```
Comment (Coly010) [inline packages/devkit/src/executors/parse-target-string.ts:16]: this is probably safe to remove :)

**Review by Coly010 — COMMENTED** [review]
Code looks good, let me check it out locally and do a test or two :)

**Comment by Coly010** [comment]
When building locally I had a TS error come from `update-12-3-0/update-storybook` migration In the angular package. 

```bash
Error: packages/angular/src/migrations/update-12-3-0/update-storybook.ts:39:11 - error TS2367: This condition will always return 'true' since the types '"@storybook/react" | "@storybook/react-native" | "@storybook/html" | "@storybook/web-components" | "@storybook/vue" | "@storybook/vue3" | "@storybook/svelte"' and '"@storybook/angular"' have no overlap.

39       if (options.uiFramework !== '@storybook/angular') {
```

Commenting that migration out, I can build. Which let me test it out.

And it works well on a newly generated angular app! :fire:

And it also works for MFE apps where we end up changing the default builder etc.

Awesome :)

**Comment by Coly010** [comment]
This looks good on my side!

There's a conflict the docs map, but otherwise this looks good!

**Review by Coly010 — APPROVED** [review]

**Review by rarmatei — APPROVED** [review]

**Pull request merged**